### DISCUSSION TITLE
Where does "Error: tdms_place.tcl, 35 cannot create std::vector larger than max_size()"

### DISCUSSION THREAD

**Discussion by oharboe** [The-OpenROAD-Project/OpenROAD#3362] [General]
I want to debug it, but I searched the code and couldn't find this error message in the source.

```
[INFO GPL-0003] SiteSize: 54 270
[INFO GPL-0004] CoreAreaLxLy: 2160 2160
[INFO GPL-0005] CoreAreaUxUy: 25922160 25922160
Error: tdms_place.tcl, 35 cannot create std::vector larger than max_size()
Command exited with non-zero status 1
```

**Comment by maliberty** [comment]
My guess is this is an STL exception and not in the OR code.  In gdb try 'catch throw' to see when it is being thrown.

**Reply by QuantamHD** [reply]
Super indicative of negative size requested

**Comment by oharboe** [comment]
This is odd... 48000*96000 = 4608000000. Ah. overflow.

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/ca384a53-04da-42fd-9039-014b3bdd89f9)

![image](https://github.com/The-OpenROAD-Project/OpenROAD/assets/2798822/a0156546-48aa-4c6c-84a1-ee9b89647226)


```
D$ git diff
diff --git a/src/gpl/src/placerBase.cpp b/src/gpl/src/placerBase.cpp
index 30ff82771..4f2c10b4e 100644
--- a/src/gpl/src/placerBase.cpp
+++ b/src/gpl/src/placerBase.cpp
@@ -1002,8 +1002,8 @@ void PlacerBase::initInstsForUnusableSites()
 {
   dbSet<dbRow> rows = db_->getChip()->getBlock()->getRows();
 
-  int siteCountX = (die_.coreUx() - die_.coreLx()) / siteSizeX_;
-  int siteCountY = (die_.coreUy() - die_.coreLy()) / siteSizeY_;
+  long siteCountX = (die_.coreUx() - die_.coreLx()) / siteSizeX_;
+  long siteCountY = (die_.coreUy() - die_.coreLy()) / siteSizeY_;
 
   enum PlaceInfo
   {
```

gives me:

```
[INFO GPL-0002] DBU: 1000
[INFO GPL-0003] SiteSize: 54 270
[INFO GPL-0004] CoreAreaLxLy: 2160 2160
[INFO GPL-0005] CoreAreaUxUy: 25922160 25922160
Error: out of memory.
```

**Reply by oharboe** [reply]
When I run `make floorplan_info` with https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/pull/1087/ I get:

```
make floorplan_info
[INFO][FLOW] Using platform directory ./platforms/asap7
[INFO-FLOW] ASU ASAP7 - version 2
Default PVT selection: BC
[INFO][FLOW] Invoked hierarchical flow.
Block Element needs to be hardened.
DIE_AREA=0 0 25924.320000000003 25924.320000000003
CORE_AREA=2.16 2.16 25922.160000000003 25922.160000000003
```

So the size is not what I expected.... There is obviously a bug in my dynamic calculation of the die/core area.

**Reply by oharboe** [reply]
So 25000 * 25000 is ca. 625mm^2. While I know that some parts are that big, I think this is in a range that is not currently relevant for OpenROAD...

I do think changing from "int" to "long" improved the error message a hair.

It's easier to surmise that the die area is too big when you get out of memory than if you get that max_size() message.

https://github.com/The-OpenROAD-Project/OpenROAD/pull/3363
